Description: GnuTLS 3.4 compat.
 Use GnuTLS priority NORMAL:+CTYPE-OPENPGP instead of deprecated (and removed in GnuTLS 3.4) function gnutls_certificate_type_set_priority.
Author: Andreas Metzler <ametzler@debian.org>
Bug-Debian: https://bugs.debian.org/624056
Origin: vendor
Forwarded: no
Last-Update: 2015-07-25

--- aiccu-20070115.orig/common/common.c
+++ aiccu-20070115/common/common.c
@@ -271,8 +271,6 @@ TLSSOCKET sock_alloc(void);
 TLSSOCKET sock_alloc(void)
 {
 #ifdef AICCU_GNUTLS
-	/* Allow connections to servers that have OpenPGP keys as well */
-	const int	cert_type_priority[3] = { GNUTLS_CRT_X509, GNUTLS_CRT_OPENPGP, 0 };
 	int		ret;
 #endif /* AICCU_GNUTLS*/
 
@@ -297,11 +295,10 @@ TLSSOCKET sock_alloc(void)
 	}
 
 	/* Use default priorities */
-	gnutls_set_default_priority(sock->session);
-	/* XXX: Return value is not documented in GNUTLS documentation! */
-
-	gnutls_certificate_type_set_priority(sock->session, cert_type_priority);
-	/* XXX: Return value is not documented in GNUTLS documentation! */
+	/* Allow connections to servers that have OpenPGP keys as well */
+	gnutls_priority_set_direct(sock->session, 
+			"NORMAL:+CTYPE-OPENPGP",
+			NULL);
 
 	/* Configure the x509 credentials for the current session */
 	gnutls_credentials_set(sock->session, GNUTLS_CRD_CERTIFICATE, g_aiccu->tls_cred);
